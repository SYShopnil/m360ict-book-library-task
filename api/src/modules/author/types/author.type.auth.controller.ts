export interface IAuthorLoginBOdy {
  password: string;
  email: string;
}
export interface IPayloadDataOfAccessToken {
  id: number;
  email: string;
}
