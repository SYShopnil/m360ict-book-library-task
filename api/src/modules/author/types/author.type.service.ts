import { IAuthor } from '../../../type/entity';

export interface IGetAllAuthors {
  authors: IAuthor[];
  total: number;
  totalPages: number;
}

export interface IRGetAuthorById extends IAuthor {
  email: string;
}
export interface IRGetAuthorByEmail extends IAuthor {
  email: string;
  password: string;
}
