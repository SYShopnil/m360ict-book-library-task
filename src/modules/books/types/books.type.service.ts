import { IBook } from '../../../type/entity';

export interface GetBooksOptions {
  page: number;
  pageSize: number;
  title?: string;
  author?: string;
}

export interface IRGetBookById extends IBook {}
export interface IRGetBooksByAuthor extends IBook {
  author_name: string;
  authors_bio: string;
}
