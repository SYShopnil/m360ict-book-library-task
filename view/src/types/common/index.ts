export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  cLoginFormWithSubmit = "cLoginFormWithSubmit",
  cLogoutContainer = "cLogoutContainer",
  rUserList = "rUserList",
  SHeaderMain = "SHeaderMain",
  SProductSection = "SProductSection",
  SProfile = "SProfile",
  CPaginationTrack = "CPaginationTrack",
  RSingleProduct = "RSingleProduct",
  RTableBody = "RTableBody",
  SProductCard = "SProductCard",
  CProductSearchBarContainer = "CProductSearchBarContainer",
}
