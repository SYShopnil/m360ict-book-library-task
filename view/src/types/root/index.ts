export type { IIconStore } from "./_icon";
export { IconName as IconEnums } from "./_icon";
export type { IButton, INormalButton, IRedirectButton } from "./Button";
export { BtnColorSchema } from "./Button";
