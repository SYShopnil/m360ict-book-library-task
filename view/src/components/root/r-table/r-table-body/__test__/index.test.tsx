import { render, screen } from "@testing-library/react";
import { RTableBody } from "..";
import { EDataTestId } from "@src/types/common";

describe("When Table Body has rendered", () => {
  it("Expect render successfully", async () => {
    // const { getByTestId } = render(
    //   await RTableBody({
    //     users: [
    //       {
    //         email: "sadmanishopnil@gmail.com",
    //         password: "123456789",
    //         userName: "John Doe",
    //         profilePicLink: "/assert/default-profile.png",
    //         userType: "user",
    //         gender: "male",
    //       },
    //       {
    //         email: "sadmanishopnil1@gmail.com",
    //         password: "123456789",
    //         userName: "Jane Smith",
    //         profilePicLink: "/assert/default-profile.png",
    //         userType: "admin",
    //         gender: "male",
    //       },
    //       {
    //         email: "xyz@gmail.com",
    //         password: "123456789",
    //         userName: "Harry Rose",
    //         profilePicLink: "/assert/default-profile.png",
    //         userType: "user",
    //         gender: "female",
    //       },
    //     ],
    //   })
    // );
    // const myElement = getByTestId(EDataTestId.RTableBody);
    const myElement = screen.getByTestId(EDataTestId.RTableBody);
    expect(myElement).toMatchSnapshot();
  });
});
