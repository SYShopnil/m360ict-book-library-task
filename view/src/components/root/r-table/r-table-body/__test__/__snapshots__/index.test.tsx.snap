// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Table Body has rendered Expect render successfully 1`] = `
<div
  data-testid="RTableBody"
>
  <div
    class="flex bg-white border-b border-gray-300"
  >
    <div
      class="w-3/6 text-center"
    >
      sadmanishopnil@gmail.com
    </div>
    <div
      class="w-1/6 text-center"
    >
      John Doe
    </div>
    <div
      class="w-1/6 text-center"
    >
      <img
        alt="Profile Pic"
        class="w-10 h-10 mx-auto rounded-full"
        src="/assert/default-profile.png"
      />
    </div>
    <div
      class="w-1/6 text-center"
    >
      user
    </div>
    <div
      class="w-1/6 text-center"
    >
      male
    </div>
  </div>
  <div
    class="flex bg-gray-100 border-b border-gray-300"
  >
    <div
      class="w-3/6 text-center"
    >
      sadmanishopnil1@gmail.com
    </div>
    <div
      class="w-1/6 text-center"
    >
      Jane Smith
    </div>
    <div
      class="w-1/6 text-center"
    >
      <img
        alt="Profile Pic"
        class="w-10 h-10 mx-auto rounded-full"
        src="/assert/default-profile.png"
      />
    </div>
    <div
      class="w-1/6 text-center"
    >
      admin
    </div>
    <div
      class="w-1/6 text-center"
    >
      male
    </div>
  </div>
  <div
    class="flex bg-white border-b border-gray-300"
  >
    <div
      class="w-3/6 text-center"
    >
      xyz@gmail.com
    </div>
    <div
      class="w-1/6 text-center"
    >
      Harry Rose
    </div>
    <div
      class="w-1/6 text-center"
    >
      <img
        alt="Profile Pic"
        class="w-10 h-10 mx-auto rounded-full"
        src="/assert/default-profile.png"
      />
    </div>
    <div
      class="w-1/6 text-center"
    >
      user
    </div>
    <div
      class="w-1/6 text-center"
    >
      female
    </div>
  </div>
</div>
`;
